scripts/tests/provisioning/test.expect

```diff
@@ -34,20 +34,22 @@ expect {
 
 set timeout 90
 # configure opkg
-expect -re "$USER@.*# $" {
-	send "touch /etc/opkg/testing.conf\r"
-	set feed_index 0
-	foreach feed_uri $ipk_feeds {
-		send -- "echo src test-feed.$feed_index http://$vm_host_ip:8990/$feed_uri >>/etc/opkg/testing.conf\r"
-		set feed_index [expr $feed_index + 1]
+expect {
+	-re "$USER@.*# $" {
+		send "touch /etc/opkg/testing.conf\r"
+		set feed_index 0
+		foreach feed_uri $ipk_feeds {
+			send -- "echo src test-feed.$feed_index http://$vm_host_ip:8990/$feed_uri >>/etc/opkg/testing.conf\r"
+			set feed_index [expr $feed_index + 1]
+		}
+		send "cat /etc/opkg/testing.conf\r"
+		send "opkg update\r"
+		send "opkg install init-restore-mode-ptest\r"
+	}
+	timeout {
+		send_user "TIMEOUT waiting for opkg to configure and ptests to install.\r"
```

Comment:
Isn't this expect clause waiting for login to complete so it can **send** the opkg commands rather than waiting **for** the opkg commands to complete? I think this same issue applies to each timeout clause below.

Reply:
No. The `timeout` clause applies to the time it takes to execute all actions within the parent `expect` clause, not the time between this clause and the previous.

When I artificially cause the `configure opkg` clause to timeout, for example, you get this:
```bash
...
NI-Unknown-52542d724e76 login: root

TIMEOUT waiting for opkg to configure and ptests to install.
INFO: Feed server (2320) terminated.
removed '/tmp/test_provisioning.nDsHD/feed.0'
removed '/tmp/test_provisioning.nDsHD/feed.1'
removed '/tmp/test_provisioning.nDsHD/feed.2'
/home/usr0/projects/ni-central/src/rtos/rtosimagesd/rtos_nilinuxrt/source/nilrt/scripts/tests/provisioning/test_provisioning.sh: line 110:  2320 Terminated              python3 -m http.server $IPK_FEED_PORT  (wd: /tmp/test_provisioning.nDsHD)
make[2]: *** [objects/test_provisioning/linuxU/x64/gcc-4.7-oe/release/ni-linux-rt-x64-qemu/provisioning.log] Error 1

```

Reply:
^ This is wrong. See Patch V5 below.